common/client.go

```diff
@@ -38,24 +38,22 @@ func (c *DatabricksClient) Configure() error {
 	c.configureHTTPCLient()
 	c.AzureAuth.databricksClient = c
 	c.userAgent = UserAgent()
-	err := c.findAndApplyAuthorizer()
-	if err != nil {
-		return err
-	}
-	if c.Host != "" && !(strings.HasPrefix(c.Host, "https://") || strings.HasPrefix(c.Host, "http://")) {
-		// azurerm_databricks_workspace.*.workspace_url is giving URL without scheme
-		// so that is why this line is here
-		c.Host = "https://" + c.Host
-	}
 	// TODO: fix it once bigger boom is done
 	//c.commandExecutor = c.Commands()
 	return nil
 }
 
-func (c *DatabricksClient) findAndApplyAuthorizer() error {
-	if c.authVisitor != nil {
-		return nil
+// ConfigureWithAuthentication configures the client to work
+// and validates if an authentication mechanism is present
+func (c *DatabricksClient) ConfigureWithAuthentication() error {
```

Comment:
Do not introduce a public method just to be used in singe unit test file. 

Please revert this `Configure` method rename, because it also deviates behaviour from the one specified in `Provider#Configure` - e.g. never called in production flow.